src/lib.rs

```diff
@@ -51,80 +51,83 @@ extern "C" {
     fn gpioSetWatchdog(user_gpio: u32, timeout: u32) -> i32; //
 }
 
+#[derive(Debug)]
+pub struct Pigpio {
+    pub version: i32,
+}
 
-
-
-/// Initializes the library.
-///
-/// Initialize must be called before using the other library functions with some exceptions not yet wrapped.
-pub fn initialize() -> GpioResponse {
-    let result = unsafe { gpioInitialise() };
-    match result {
-        INIT_FAILED => Err("Initialize failed".to_string()),
-        _ => Ok(result as u32)
+impl Drop for Pigpio {
+    /// Terminates the library.
+    ///
+    /// This function resets the used DMA channels, releases memory, and terminates any running threads.
+    fn drop(&mut self) {
+        unsafe { gpioTerminate() };
     }
-
 }
 
-/// Terminates the library.
-///
-/// Call before program exit.
-/// This function resets the used DMA channels, releases memory, and terminates any running threads.
-pub fn terminate() {
-    unsafe { gpioTerminate() };
-}
+impl Pigpio {
+    /// Initializes the library.
+    pub fn new() -> Result<Pigpio, &'static str> {
```

Comment:
Shouldn't this use GpioResponse?

Reply:
But this has the Pigpio stuct as value and GpioResponse a u32. I changed to use String in Error case, like GpioResponse